lib/routes/user/home/payments_filter.dart

```diff
@@ -83,11 +88,28 @@ class PaymentsFilter extends StatelessWidget {
                     ),
                   );
                 }).toList(),
-                onChanged: (value) => _onFilterChanged(value)),
+                onChanged: (value) => _onFilterChanged(value,_startDate,_endDate)),
           ),
         ),
-      ),
     );
-    return Row(children: children);
+    return new Column(mainAxisSize: MainAxisSize.max, mainAxisAlignment: MainAxisAlignment.start, crossAxisAlignment: CrossAxisAlignment.start
+      ,children: <Widget>[new Padding(padding: EdgeInsets.only(top: 16.0),child: Row(children: children)), _buildDateFilterChip(_startDate,_endDate)],);
+  }
+
+  _buildDateFilterChip(DateTime startDate,DateTime endDate){
+    if(_startDate != null && _endDate != null){
+      return new Padding(padding: EdgeInsets.only(left: 16.0),child: Chip(label: Text(_formatFilterDateRange(_startDate,_endDate)),onDeleted: (){ _onFilterChanged(_filter,null,null);} ,));
+    }
+    return Container();
+  }
+
+  String _formatFilterDateRange(DateTime startDate,DateTime endDate) {
```

Comment:
This function is also duplicated in the pos and here. See my comment on CalendarFilter regarding DateUtils. it maybe belong there as well.